### PR TITLE
Minor improvement to provider annotations

### PR DESCRIPTION
**Pull request by lucyleeow**
# References and relevant issues

Result of [this](https://napari.zulipchat.com/#narrow/channel/212875-general/topic/giving.20Viewer.20to.20plugins) discussion on zulip.

# Description

* Clarify napari adds all types from the listed *modules* for annotation
* Minor amendments

### CODE REVIEW

File: docs/developers/architecture/app_model.md [hunk 1]
```diff
@@ -346,7 +346,7 @@ dependency to be injected.
 Result processing allows you to process the return value of the command function at
 execution time, based on command return type annotations.
 For example, you may wish to automatically add layer data output from a command
-to the viewer. It is performed by `processors`, functions that accept an instance of a
```
Comment (lucyleeow) [inline docs/developers/architecture/app_model.md:349]: Removed because all other mentors of 'processors' do not use backticks

**Comment by lucyleeow** [comment]
cc @jni

**Review by DragaDoncila — APPROVED** [review]
Thanks for the clarifications @lucyleeow

**Comment by lucyleeow** [comment]
I'm going to merge this

**Pull request merged**